enhancements/installer/aws-ingress-nlb-security-group.md

```diff
@@ -0,0 +1,501 @@
+  - Introduce a new field (`managedSecurityGroup`) in the NLB provider parameters within the IngressController API.
+  - Controller creates the Service for the default router with the annotation `service.beta.kubernetes.io/aws-load-balancer-type: nlb` and the new annotation `service.beta.kubernetes.io/aws-load-balancer-managed-security-group: "true"` when the `managedSecurityGroup` flag is enabled.
+- Cloud Controller Manager (CCM):
+  - Enable support for the new annotation `service.beta.kubernetes.io/aws-load-balancer-managed-security-group: "true"` when creating a service type Load Balancer with `aws-load-balancer-type: nlb`.
+  - When the annotation is present, the CCM will:
+    - Create a new Security Group instance for the NLB.
+    - Create Ingress and Egress rules in the Security Group based on the NLB Listeners and Target Groups' ports. Initially, egress rules should be restricted to the necessary ports for backend communication (traffic and health check ports).
+    - Pass the ID of the newly created Security Group during the NLB creation via the AWS API.
+    - Delete the Security Group when the corresponding service is deleted.
+  - Enhance existing tests for the Load Balancer component in the CCM to include scenarios with the new annotation.
+
+Risk:
+- CCM/upstream:
+  - SG management increases controller complexity and scenarios to validate.
+  - API changes (new annotation).
+  - More changes upstream increase the consensus/approvals required, especially for new features in service LB on CCM (prefer ALBC in the long term).
+  - More changes in CCM to create and manage the SG lifecycle.
+- CCM/downstream:
+  - More complex to maintain downstream if not fully upstreamed.
+
+Day-2:
+- Self-managed: Patch CIO.
+
+**Phase 2A: Create support on ROSA Classic**
+
+Goals:
+- TBD: Investigate how ROSA Classic (Hive?) reads and reacts to the new option in `install-config`: `platform.aws.ingressController.managedSecurityGroup`. Ensure the CIO manifests are generated correctly.
+
+Day-2:
+- Managed Services: TBD: Patch CIO.
+
+**Phase 2B: Create support on ROSA HCP**
+
+Goals:
+- TBD: Investigate how Hypershift sets the Service Annotation for Security Group management when launching CIO for Hosted Control Planes.
+- TBD: Explore the HCP flow to validate if the self-managed proposed covers it or if specific adjustments are needed.
+
+Day-2:
+- Managed Services: TBD: Patch CIO.
+
+### Workflow Description
+
+#### OpenShift Self-managed
+
+- User:
+  - Creates `install-config.yaml` enabling the use of Security Group, example `platform.aws.ingressController.securityGroupEnabled`, **when** `lbType=NLB` (already exists).
+```yaml
+# install-config.yaml
+platform:
+  aws:
+    region: us-east-1
+    lbType: NLB                   <-- deprecate by platform.aws.ingressController.loadBalancerType?
+    ingressController:            <-- proposing to aggregate CIO configurations
+      securityGroupEnabled: True  <-- new field
```

Comment:
What if I want to have different security groups for ingress vs the rest of the cluster? Is that possible?

Do we need the option for this to be automatic (use the same as you'd expect for default) but also a BYO option where users can specify specific SG IDs to be used?